viewer/packages/pointclouds/src/PointCloudFactory.ts

```diff
@@ -38,30 +42,34 @@ export class PointCloudFactory {
     }
 
     if (geometry.cylinder) {
-      return new CylinderPrimitive(geometry.cylinder.centerA, geometry.cylinder.centerA, geometry.cylinder.radius);
+      const centerA = new THREE.Vector3().fromArray(geometry.cylinder.centerA);
+      const centerB = new THREE.Vector3().fromArray(geometry.cylinder.centerB);
+      return new CylinderPrimitive(centerA, centerB, geometry.cylinder.radius);
     }
 
     throw Error('Annotation geometry type not recognized');
   }
 
-  private async getAnnotations(modelIdentifier: CdfModelIdentifier): Promise<BoundingVolume[]> {
+  private async getAnnotations(modelIdentifier: CdfModelIdentifier): Promise<CdfPointCloudObjectAnnotation[]> {
     const modelAnnotations = await this._sdkPlayground!.annotations.list({
       filter: {
         // @ts-ignore
         annotatedResourceType: 'threedmodel',
         annotatedResourceIds: [{ id: modelIdentifier.modelId }]
-      }
-    });
+      },
+      limit: 1000
+    }).autoPagingToArray({ limit: Infinity });
 
-    const bvs = modelAnnotations.items.map(annotation => {
+    const bvs = modelAnnotations.map(annotation => {
       const region = (annotation.data as any).region.map((geometry: any) => {
         return this.annotationGeometryToLocalGeometry(geometry);
       });
 
       return {
         annotationId: annotation.id,
+        assetId: annotation.annotatedResourceId,
         region
-      } as BoundingVolume;
+      } as CdfPointCloudObjectAnnotation;
```

Comment:
Can this be done more type safe? Using non-playground SDK?

If SDK doesn't have the correct types, add a task for that and make sure we have it on the TODO-list.

Reply:
The types are not ready in the SDK, @Savokr has taken on the task of adding them there, but not sure when it's ready. Changing these types to the SDK ones are tracked as a bullet point here: https://cognitedata.atlassian.net/browse/REV-374